### PR TITLE
Add 'argo' tag to wandb-tags for training/offline diags

### PR DESCRIPTION
**Pull request by oliverwm1**
This will make it easier to connect `wandb` runs with models found by searching on GCS (e.g. using `artifacts ls`).

The PR also sets the WANDB_JOB_TYPE environment variable for the training and offline diags argo workflows.

Coverage reports (updated automatically):
- test_unit: [83%](https://output.circle-artifacts.com/output/job/c24c5fb5-ee17-42fe-ad90-f79361ffb10b/artifacts/0/tmp/coverage/htmlcov-test_unit/index.html)

### CODE REVIEW

**Comment by oliverwm1** [comment]
I launched an e2e test locally to confirm this works. See example wandb run: https://wandb.ai/ai2cm/argo-default/runs/195zcqjg/overview

File: workflows/argo/offline-diags.yaml [hunk 1]
```diff
@@ -43,6 +43,8 @@ spec:
             value: "{{inputs.parameters.wandb-project}}"
           - name: WANDB_RUN_GROUP
             value: "{{inputs.parameters.wandb-group}}"
+          - name: WANDB_JOB_TYPE
+            value: offline_report
```
Comment (oliverwm1) [inline workflows/argo/offline-diags.yaml:47]: chosen to be consistent with the job_type defined in `StepMetadata` in offline report code.

File: workflows/argo/training.yaml [hunk 2]
```diff
@@ -41,6 +41,8 @@ spec:
             value: "{{inputs.parameters.wandb-project}}"
           - name: WANDB_RUN_GROUP
             value: "{{inputs.parameters.wandb-group}}"
+          - name: WANDB_JOB_TYPE
+            value: training
```
Comment (oliverwm1) [inline workflows/argo/training.yaml:45]: consistent with job_type in `StepMetadata` defined in fv3fit.train entrypoint.

**Comment by oliverwm1** [comment]
Seems to be working: https://wandb.ai/ai2cm/argo-default/runs/1p7b9e4m/overview?workspace=user-oliverwm

**Review by AnnaKwa — APPROVED** [review]
LGTM

**Pull request merged**